leakcanary-android-core/src/main/java/leakcanary/LeakCanary.kt

```diff
@@ -169,7 +169,14 @@ object LeakCanary {
      * Deprecated: This is a no-op, set a custom [leakingObjectFinder] instead.
      */
     @Deprecated("This is a no-op, set a custom leakingObjectFinder instead")
-    val useExperimentalLeakFinders: Boolean = false
+    val useExperimentalLeakFinders: Boolean = false,
+
+    /**
+     * Whether the LeakTrace should be wrapped (to 80 chars) or not, when shared
+     *
+     * Defaults to true.
+     */
+    val wrapLeakTraceWhenSharing: Boolean = true
```

Comment:
I'm not convinced we need to increase the API surface of the config for this. I expect that the decision to wrap or not is use case based rather than an overall setting for all of LeakCanary.

From an earlier comment:

> provide another method (e.g. toWrappedString()) that takes a width parameter with a default value of 80, and then update callsites inside LeakCanary to use that. The result is that anyone currently using toString will not have any breaking change.

So, to summarize:

* The behavior does not change when calling `HeapAnalysisSuccess.toString()`, we still get no wrapping
* A new method appears: `toWrappedString(width = 80)`
* Current call sites of `HeapAnalysisSuccess.toString()` are updated to call toWrappedString instead.

What do you think?

Reply:
I added this config based on cketti's suggestion in one of the comments. I also thought it could provide a quick workaround  if the wrapping algorithm didn't work well on some unexpected cases. However, I understand that we should be careful when increasing the API surface of the config. I will remove it.

That being said, with this change, `HeapAnalysisSucess.toString()` implementation is not impacted. In fact, the wrapping is applied externally on two specific use cases: `share` and `share_stack_overflow` in `LeakScreen`.

The advantage of this, compared to adding a `toWrappedString` to `HeapAnalysisSuccess`,  is that this is presentation logic and I thought it made more sense that it stays closer to the presentation code rather than adding it deep down in `shark`.

What do you think? Do you see other call sites? Do you see advantages in having the wrapping logic belong to `shark` instead?